tests/network/cifs.pm

```diff
@@ -0,0 +1,105 @@
+    $self->select_serial_terminal;
+    add_suseconnect_product(get_addon_fullname('phub')) if is_sle;    # samba-client requires package hub
+    zypper_call('in cifs-utils samba-client nmap');
+    # Use local samba server, if defined or if defined SMB server is not accessible
+    my $is_local = get_var("CIFS_TEST_REMOTE") eq 'local';
+    if ($is_local || script_run("nmap -p 139,445 $smb_remote | grep open") != 0) {
+        my $reason = $is_local ? "defined by CIFS_TEST_REMOTE" : "$smb_remote unreachable";
+        record_info("local samba server", "Using a local samba server (defined by CIFS_TEST_REMOTE)");
+        setup_local_server();
+        $smb_remote = "127.0.0.1";
+    }
+    script_run("smbclient -m SMB2 -L $smb_domain -I $smb_remote -U guest -N");
+    assert_script_run("smbclient -m SMB2 -L $smb_domain -I $smb_remote -U guest -N | grep 'Disk' | grep -i 'currywurst'");
+    assert_script_run("smbclient -m SMB3 -L $smb_domain -I $smb_remote -U guest -N | grep 'Disk' | grep -i 'currywurst'");
+    # Test CIFS mount
+    assert_script_run("mkdir -p /mnt/{currywurst,filedrop}");
+    my $options = "username=geekotest,password=nots3cr3t";
+    # Test mount with the different version types
+    if (is_sle(">12-SP3")) {
+        foreach my $version (@versions) {
+            assert_script_run("mount -t cifs -o $options,vers=$version //$smb_remote/currywurst /mnt/currywurst");
+            assert_script_run("umount /mnt/currywurst");
+        }
+    } else {
+        $options .= ",vers=2.1";    # needed for SLES12-SP3 and below
+    }
+    assert_script_run("mount -t cifs -o $options //$smb_remote/currywurst /mnt/currywurst");
+    assert_script_run("mount -t cifs -o $options //$smb_remote/filedrop /mnt/filedrop");
+    # Check if test files are there
+    assert_script_run("stat /mnt/currywurst/Recipe.txt");
+    my $filename = random_string(8);    # random filename to prevent race conditions on the server
+    assert_script_run("cp /mnt/currywurst/Recipe.txt /mnt/filedrop/$filename");
+    assert_script_run("umount /mnt/currywurst");
+    assert_script_run("! stat /mnt/currywurst/Recipe.txt");
+    assert_script_run("stat /mnt/filedrop/$filename");
+    assert_script_run("md5sum /mnt/filedrop/$filename > Recipe.txt.md5sum");
+    # Check if file are in the filedrop after a remount
+    assert_script_run("umount /mnt/filedrop");
+    assert_script_run("mount -t cifs -o $options,ro //$smb_remote/filedrop /mnt/filedrop");
+    assert_script_run("stat /mnt/filedrop/$filename");
+    assert_script_run("md5sum -c Recipe.txt.md5sum");
+    assert_script_run("! rm /mnt/filedrop/$filename");
+    assert_script_run("mount -t cifs -o $options,remount,rw /mnt/filedrop");
+    assert_script_run("rm /mnt/filedrop/$filename");
+}
+
+
+sub post_fail_hook {
+    script_run("umount /mnt/{currywurst,filedrop}");
+    script_run("rmdir /mnt/{currywurst,filedrop}");
+}
+
+sub post_run_hook {
+    script_run("rmdir /mnt/{currywurst,filedrop}");
```

Comment:
Nice work Felix! (I will try the recipe too ^^)
One small thing that I noticed in the VR is when the `post_run_hook` is performed, it returns `rmdir: failed to remove '/mnt/filedrop': Device or resource busy`, so just make sure that the medium can be removed ;)

Apart from this minor notice, lgtm!

Reply:
Nice catch! Should add a `umount` there too. Thanks Anna :-)